### PR TITLE
feat: add dummy helm releases

### PR DESCRIPTION
**Pull request by vespian**
This PR is meant to unblock https://jira.d2iq.com/browse/D2IQ-82657 on one hand and on the other - enable pararelising the upgrades work as once it gets merged it can be worked on independently.

The idea behind this PR is to add dummy helm releases that will replace dynamic helm releases generated by:
* federation/pkg/controllers/kommandercluster_prometheus_proxy.go
* federation/pkg/controllers/kommandercluster_observer_controller.go

This way the charts parsing logic that populates helm-mirror/chartmuseum in https://github.com/mesosphere/kommander-cli/pull/298 will be able to determine versions of charts to download while still letting controllers create helm releases dynamically. The only difference is going to be that the controllers will be creating Flux kustomizations that point to these templates instead.

### CODE REVIEW

**Review by gracedo — COMMENTED** [review]
> The only difference is going to be that the controllers will be creating Flux kustomizations that point to these templates instead.

So there will be a corresponding kommander pr updating the controllers? Can we review those changes (when ready) alongside this PR before merging please? I think we should be able to update the k-apps ref https://github.com/mesosphere/kommander/blob/main/Makefile#L60 to point to these changes here to test them together

**Comment by makkes** [comment]
@mesosphere-mergebot test all

**Comment by makkes** [comment]
@mesosphere-mergebot test all

**Comment by makkes** [comment]
rebased branch to make pre-commit pass.

File: services/kommander/0.1.0/dynamic-helmreleases/README.txt
Comment (makkes) [inline services/kommander/0.1.0/dynamic-helmreleases/README.txt:2]: Can we add some explanation here on what "dynamically generated" HelmReleases are and why we need them, please? This will help our future selves recall what this dir is about.

Comment (vespian) [reply services/kommander/0.1.0/dynamic-helmreleases/README.txt]: Comment added.

**Comment by vespian** [comment]
@makkes Comment added, can you re-review?

**Comment by vespian** [comment]
@makkes Comment added, can you re-review?

**Review by makkes — APPROVED** [review]

**Comment by vespian** [comment]
> Can we review those changes (when ready) alongside this PR before merging please? I think we should be able to update the k-apps ref https://github.com/mesosphere/kommander/blob/main/Makefile#L60 to point to these changes here to test them together

@gracedo This PR only unblocks https://jira.d2iq.com/browse/D2IQ-82657 The files themselves are not used yet, only there to make creating charts-bundles happy. So there is absolutely no risk here with merging it - just treat them as plain text files for now. 

The change to controllers themselves will happen in https://jira.d2iq.com/browse/D2IQ-83144 and it is quite possible that these files will evolve even more as there is some back-and-forth regarding the design.

File: services/kommander/0.1.0/dynamic-helmreleases/cluster-observer/cluster-observer.yaml
Comment (mikolajb) [inline services/kommander/0.1.0/dynamic-helmreleases/cluster-observer/cluster-observer.yaml:5]: `overriden` ➡️ `overridden`

**Review by mikolajb — APPROVED** [review]

**Comment by gracedo** [comment]
> The files themselves are not used yet, only there to make creating charts-bundles happy. So there is absolutely no risk here with merging it - just treat them as plain text files for now.

@vespian sgtm if there is no harm in merging these files without the corresponding controller, I'll unblock 👍🏻

File: services/kommander/0.1.0/dynamic-helmreleases/cluster-observer/cluster-observer.yaml
Comment (vespian) [reply services/kommander/0.1.0/dynamic-helmreleases/cluster-observer/cluster-observer.yaml]: Done.

**Pull request merged**
